discovery/uyuni/uyuni.go

```diff
@@ -0,0 +1,343 @@
+		"system.monitoring.listEndpoints",
+		[]interface{}{token, systemIDs}, &endpointInfos)
+	if err != nil {
+		return nil, err
+	}
+	return endpointInfos, err
+}
+
+// NewDiscovery returns a new file discovery for the given paths.
+func NewDiscovery(conf *SDConfig, logger log.Logger) *Discovery {
+	d := &Discovery{
+		interval: time.Duration(conf.RefreshInterval),
+		sdConfig: conf,
+		logger:   logger,
+	}
+	d.Discovery = refresh.NewDiscovery(
+		logger,
+		"uyuni",
+		time.Duration(conf.RefreshInterval),
+		d.refresh,
+	)
+	return d
+}
+
+func (d *Discovery) getEndpointLabels(
+	endpoint endpointInfo,
+	systemGroupIDs []systemGroupID,
+	networkInfo networkInfo,
+) model.LabelSet {
+
+	var hostname string
+	var addr string
+	managedGroupNames := getSystemGroupNames(systemGroupIDs)
+	if len(networkInfo.PrimaryFQDN) > 0 {
+		hostname = networkInfo.PrimaryFQDN
+	} else {
+		hostname = networkInfo.Hostname
+	}
+	if endpoint.Port > 0 {
+		addr = fmt.Sprintf("%s:%d", hostname, endpoint.Port)
+	} else {
+		addr = hostname
+	}
+
+	result := model.LabelSet{
+		model.AddressLabel:                       model.LabelValue(addr),
+		uyuniMetaLabelPrefix + "minion_hostname": model.LabelValue(hostname),
+		uyuniMetaLabelPrefix + "system_id":       model.LabelValue(fmt.Sprintf("%d", endpoint.SystemID)),
+	}
+
+	if len(managedGroupNames) > 0 {
+		result[uyuniMetaLabelPrefix+"groups"] = model.LabelValue(strings.Join(managedGroupNames, ","))
+	}
+	if len(endpoint.EndpointName) > 0 {
+		result[uyuniMetaLabelPrefix+"endpoint_name"] = model.LabelValue(endpoint.EndpointName)
+	}
+	if len(endpoint.ExporterName) > 0 {
+		result[uyuniMetaLabelPrefix+"exporter"] = model.LabelValue(endpoint.ExporterName)
+	}
+	if len(endpoint.Module) > 0 {
+		result[uyuniMetaLabelPrefix+"proxy_module"] = model.LabelValue(endpoint.Module)
+	}
+	if len(endpoint.Path) > 0 {
+		result[uyuniMetaLabelPrefix+"metrics_path"] = model.LabelValue(endpoint.Path)
+	}
+	level.Debug(d.logger).Log("msg", "Configured target", "Labels", fmt.Sprintf("%+v", result))
+
+	return result
+}
+
+func getSystemGroupNames(systemGroupsIDs []systemGroupID) []string {
+	managedGroupNames := make([]string, 0, len(systemGroupsIDs))
+	for _, systemGroupInfo := range systemGroupsIDs {
+		managedGroupNames = append(managedGroupNames, systemGroupInfo.GroupName)
+	}
+
+	if len(managedGroupNames) == 0 {
+		managedGroupNames = []string{"No group"}
+	}
+	return managedGroupNames
+}
+
+func (d *Discovery) getTargetsForSystems(
+	rpcClient *xmlrpc.Client,
+	token string,
+	systemGroupIDsBySystemID map[int][]systemGroupID,
+) ([]model.LabelSet, error) {
+
+	result := make([]model.LabelSet, 0)
+
+	systemIDs := make([]int, 0, len(systemGroupIDsBySystemID))
+	for systemID := range systemGroupIDsBySystemID {
+		systemIDs = append(systemIDs, systemID)
+	}
+
+	endpointInfos, err := getEndpointInfoForSystems(rpcClient, token, systemIDs)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to get endpoints information")
```

Comment:
```suggestion
		return nil, errors.Wrap(err, "unable to get endpoints information")
```

Errors are lowercased